### ISSUE TITLE
What is the `-g` option in `spark migrate` for?

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#7891]
Migration of non-default group is also executed without the `-g` option.

> -g - to chose database group, otherwise default database group will be used.
https://codeigniter4.github.io/CodeIgniter4/dbmgmt/migration.html#migrate

### How to Reproduce

```diff
--- a/app/Config/Database.php
+++ b/app/Config/Database.php
@@ -45,6 +45,47 @@ class Database extends Config
         'numberNative' => false,
     ];
 
+    public array $test1 = [
+        'DSN'          => '',
+        'hostname'     => 'localhost',
+        'username'     => 'root',
+        'password'     => '',
+        'database'     => 'ci4_test1',
+        'DBDriver'     => 'MySQLi',
+        'DBPrefix'     => '',
+        'pConnect'     => false,
+        'DBDebug'      => true,
+        'charset'      => 'utf8',
+        'DBCollat'     => 'utf8_general_ci',
+        'swapPre'      => '',
+        'encrypt'      => false,
+        'compress'     => false,
+        'strictOn'     => false,
+        'failover'     => [],
+        'port'         => 3306,
+        'numberNative' => false,
+    ];
+    public array $test2 = [
+        'DSN'          => '',
+        'hostname'     => 'localhost',
+        'username'     => 'root',
+        'password'     => '',
+        'database'     => 'ci4_test2',
+        'DBDriver'     => 'MySQLi',
+        'DBPrefix'     => '',
+        'pConnect'     => false,
+        'DBDebug'      => true,
+        'charset'      => 'utf8',
+        'DBCollat'     => 'utf8_general_ci',
+        'swapPre'      => '',
+        'encrypt'      => false,
+        'compress'     => false,
+        'strictOn'     => false,
+        'failover'     => [],
+        'port'         => 3306,
+        'numberNative' => false,
+    ];
+
     /**
      * This database connection is used when
      * running PHPUnit database tests.
```

```php
<?php

namespace App\Database\Migrations;

use CodeIgniter\Database\Migration;

class AddBlog extends Migration
{
    protected $DBGroup = 'test1';

    public function up()
    {
        $this->forge->addField([
            'blog_id' => [
                'type'           => 'INT',
                'constraint'     => 5,
                'unsigned'       => true,
                'auto_increment' => true,
            ],
            'blog_title' => [
                'type'       => 'VARCHAR',
                'constraint' => '100',
            ],
            'blog_description' => [
                'type' => 'TEXT',
                'null' => true,
            ],
        ]);
        $this->forge->addKey('blog_id', true);
        $this->forge->createTable('blog');
    }

    public function down()
    {
        $this->forge->dropTable('blog');
    }
}
```

```console
$ php spark migrate

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 02:12:42 UTC+00:00

Running all new migrations...
	Running: (App) 2023-09-04-013453_App\Database\Migrations\AddBlog
Migrations complete.
```

```console
$ php spark migrate:status

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 02:15:26 UTC+00:00

+-----------+-------------------+----------+-------+---------------------+-------+
| Namespace | Version           | Filename | Group | Migrated On         | Batch |
+-----------+-------------------+----------+-------+---------------------+-------+
| App       | 2023-09-04-013453 | AddBlog  | test1 | 2023-09-04 02:12:42 | 1     |
+-----------+-------------------+----------+-------+---------------------+-------+
```

### References

- #7846
- #5664
- #5087

**Comment by kenjis** [comment]
The migration specifications are not clear, but my understanding is as follows:
- If a non-default DB group is specified, the `$DBGroup` in the migration file must be specified.
- Migration must manage the state of the entire databases, so the migration table is always created in the default database group.
- It is not expected that a single migration file will be executed for multiple databases.

Is there something wrong? > @codeigniter4/database-team

**Comment by michalsn** [comment]
@kenjis All 3 points are valid.

`-g` is for running migrations only for specified database.

I believe the lack of this parameter should execute all migrations.

**Comment by kenjis** [comment]
Okay, then the description in the user guide is wrong?

> -g - to chose database group, **otherwise default database group will be used.**

**Comment by michalsn** [comment]
I think so.

**Comment by kenjis** [comment]
Okay, it seems the current behavior is fine. And the bug is only in the docs.

If you don't specify `-g`, migrations for all groups run.

```console
$ php spark migrate:status

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:47:49 UTC+00:00

+-----------+-------------------+----------------+-------+-------------+-------+
| Namespace | Version           | Filename       | Group | Migrated On | Batch |
+-----------+-------------------+----------------+-------+-------------+-------+
| App       | 2023-09-04-013453 | DefaultAddBlog | ---   | ---         | ---   |
| App       | 2023-09-04-013454 | Test1AddBlog   | ---   | ---         | ---   |
| App       | 2023-09-04-013455 | Test2AddBlog   | ---   | ---         | ---   |
+-----------+-------------------+----------------+-------+-------------+-------+
```
```console
$ php spark migrate

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:48:00 UTC+00:00

Running all new migrations...
        Running: (App) 2023-09-04-013453_App\Database\Migrations\DefaultAddBlog
        Running: (App) 2023-09-04-013454_App\Database\Migrations\Test1AddBlog
        Running: (App) 2023-09-04-013455_App\Database\Migrations\Test2AddBlog
Migrations complete.
```
```console
$ php spark migrate:status

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:48:04 UTC+00:00

+-----------+-------------------+----------------+---------+---------------------+-------+
| Namespace | Version           | Filename       | Group   | Migrated On         | Batch |
+-----------+-------------------+----------------+---------+---------------------+-------+
| App       | 2023-09-04-013453 | DefaultAddBlog | default | 2023-09-04 05:48:00 | 1     |
| App       | 2023-09-04-013454 | Test1AddBlog   | test1   | 2023-09-04 05:48:00 | 1     |
| App       | 2023-09-04-013455 | Test2AddBlog   | test2   | 2023-09-04 05:48:00 | 1     |
+-----------+-------------------+----------------+---------+---------------------+-------+
```

If you specify `default`:
```console
$ php spark migrate:status

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:48:10 UTC+00:00

+-----------+-------------------+----------------+-------+-------------+-------+
| Namespace | Version           | Filename       | Group | Migrated On | Batch |
+-----------+-------------------+----------------+-------+-------------+-------+
| App       | 2023-09-04-013453 | DefaultAddBlog | ---   | ---         | ---   |
| App       | 2023-09-04-013454 | Test1AddBlog   | ---   | ---         | ---   |
| App       | 2023-09-04-013455 | Test2AddBlog   | ---   | ---         | ---   |
+-----------+-------------------+----------------+-------+-------------+-------+
```
```console
$ php spark migrate -g default

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:49:02 UTC+00:00

Running all new migrations...
        Running: (App) 2023-09-04-013453_App\Database\Migrations\DefaultAddBlog
Migrations complete.
```

```console
$ php spark migrate:status

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:49:05 UTC+00:00

+-----------+-------------------+----------------+---------+---------------------+-------+
| Namespace | Version           | Filename       | Group   | Migrated On         | Batch |
+-----------+-------------------+----------------+---------+---------------------+-------+
| App       | 2023-09-04-013453 | DefaultAddBlog | default | 2023-09-04 05:49:02 | 1     |
| App       | 2023-09-04-013454 | Test1AddBlog   | ---     | ---                 | ---   |
| App       | 2023-09-04-013455 | Test2AddBlog   | ---     | ---                 | ---   |
+-----------+-------------------+----------------+---------+---------------------+-------+
```

If you specify `test2`:
```console
$ php spark migrate -g test2

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:49:21 UTC+00:00

Running all new migrations...
        Running: (App) 2023-09-04-013455_App\Database\Migrations\Test2AddBlog
Migrations complete.
```

```console
$ php spark migrate:status

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:49:23 UTC+00:00

+-----------+-------------------+----------------+---------+---------------------+-------+
| Namespace | Version           | Filename       | Group   | Migrated On         | Batch |
+-----------+-------------------+----------------+---------+---------------------+-------+
| App       | 2023-09-04-013453 | DefaultAddBlog | default | 2023-09-04 05:49:02 | 1     |
| App       | 2023-09-04-013454 | Test1AddBlog   | ---     | ---                 | ---   |
| App       | 2023-09-04-013455 | Test2AddBlog   | test2   | 2023-09-04 05:49:21 | 2     |
+-----------+-------------------+----------------+---------+---------------------+-------+
```

If you specify `test1`:
```console
$ php spark migrate -g test1

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:49:33 UTC+00:00

Running all new migrations...
        Running: (App) 2023-09-04-013454_App\Database\Migrations\Test1AddBlog
Migrations complete.
```

```console
$ php spark migrate:status

CodeIgniter v4.4.0 Command Line Tool - Server Time: 2023-09-04 05:49:34 UTC+00:00

+-----------+-------------------+----------------+---------+---------------------+-------+
| Namespace | Version           | Filename       | Group   | Migrated On         | Batch |
+-----------+-------------------+----------------+---------+---------------------+-------+
| App       | 2023-09-04-013453 | DefaultAddBlog | default | 2023-09-04 05:49:02 | 1     |
| App       | 2023-09-04-013454 | Test1AddBlog   | test1   | 2023-09-04 05:49:33 | 3     |
| App       | 2023-09-04-013455 | Test2AddBlog   | test2   | 2023-09-04 05:49:21 | 2     |
+-----------+-------------------+----------------+---------+---------------------+-------+
```

**Comment by michalsn** [comment]
Great, thanks!

**Comment by kenjis** [comment]
Closed by #7894
